### PR TITLE
Fix the error of splitting chatglm, qwen-7b-chat and baichuan-13b

### PR DESCRIPTION
**Pull request by WeldonWangwang**
### Details:

- When there are some submodels need to be merged into one model, if there is a independent submodel in the submodels, such as `chatglm, qwen-7b-chat`:
submodel A (independent )
submodel B
submodel C
B and C will be the same because they have connection(input/output), so when merge C into A after B merged, it will throw  `parameters already added` errior in add_parameters().
 So add a distinct_submodels_index set to store the distinct submodels will be merged.

- When split `baichuan-13b` into 2 parts, it will has error: Model references undeclared parameters: opset1::Parameter input_ids () -> (i64[?,?]), because there are some ops not removed correctly.

### Tickets:
 - *CVS-140527*

### CODE REVIEW

File: src/inference/src/dev/iplugin.cpp [hunk 1]
```diff
@@ -313,65 +331,109 @@ std::unordered_set<std::string> ov::get_supported_nodes(
...
-            if ((total_len < last_total_len || last_total_len == 0) && !cancel_split) {
-                last_total_len = total_len;
-                copy_set(supported, temp_supported_1);
-                copy_set(unsupported, temp_unsupported_1);
+                if ((total_len < last_total_len || last_total_len == 0) && !cancel_split) {
+                    last_total_len = total_len;
+                    copy_set(supported, temp_supported_1);
+                    copy_set(unsupported, temp_unsupported_1);
+                }
             }
```
Comment (songbell) [inline src/inference/src/dev/iplugin.cpp:437]: want to make sure, that do the changes in iplugin interface relevant to the issue of this PR?

Comment (WeldonWangwang) [reply src/inference/src/dev/iplugin.cpp:437]: Hi @songbell , update the Details of this PR, please review.

**Review by wangleis — APPROVED** [review]

**Comment by wangleis** [comment]
@ilya-lavrenov Do you have comment on change of src/inference/src/dev/iplugin.cpp?

**Comment by ilya-lavrenov** [comment]
> @ilya-lavrenov Do you have comment on change of src/inference/src/dev/iplugin.cpp?

no.

maybe in future we can move `get_supported_nodes` to a dedicated file, because it has almost nothing common with iplugin :)

**Review by songbell — APPROVED** [review]

**Pull request merged**